Detailed status assertions (#61)
peppelinux/draft-demarco-oauth-status-assertions: draft-demarco-oauth-status-attestations.md

allowing Verifiers to examine and make educated conclusions based on a
variety of scenarios.

# Assertions of Invalidity

Issuer's MAY provide an assertion of invalidity, in contrast to an assertion of validity or an error for digital credentials which have been revoked.

> This enables verifier policies to be conditioned on the presence of signed positive or negative state from the issuer, instead of the absence of information.

Reviewer: ```suggestion
# Extendibility of the Status Assertions

Status Assertions are used to represent the binary state of a Credential, such as valid or invalid, particularly in cases where a Credential has been revoked, resulting in either a validity Status Assertion or a Status Assertion Error.

However, there are instances where implementations might use this specification to characterize additional states of a Credential, such as suspension.

For instance, an Issuer MAY provide Status Assertions of invalidity.

Even if these cases are out of the scopes of this specification, these enable verifier policies to be conditioned on the presence of signed positive or negative state from the issuer, instead of the absence of information.

In these cases, in addition of the parameter defined in this specification about the Status Assertions ...
```

@OR13 the problem I see is that by extending the data set we may incur in implementations that only evaluates the `typ` for the binary state of non revoked, therefore valid.

This means that we should add another required claim with a set of extendible values and say what the verifier should do when gets a status value that it doesn't recognize. According to the OAuth phylosophy it should ignore the single parameter, while while in this case, for security reason, it should invalidate the entire assertion

Author: My mental model for status assertions is that they represent dynamic state of the credential, under control of the issuer, which can be provided after issuance. 

In my mental model, state is not limited to booleans, and it is the verifier who ultimately determines validity (or more accurately acceptability of complex state), the issuer simply provides the input to their decision process.

Revocation is a special case of boolean value, and is somewhat overloaded in the context of credentials, since keys and certificates can be revoked, as well as claims.

Author: I can't apply this suggestion, as it is outdated.

allowing Verifiers to examine and make educated conclusions based on a
variety of scenarios.

# Complex Status Assertions

Status assertions can be complex, and are not limited to simple boolean information.
This enables verifier policies to be conditioned on the presence of secured information, instead of the absence of information.
This section proposes syntax to support complex assertions.
The `validity` claim MUST be present and be either `true` or `false`.

> The `validity_reasons` MAY be present and if present MUST be an object.

Author: ```suggestion
The `validity_reasons` claim MAY be present and if present MUST be an object.
```

a deliberate effort to balance security and privacy needs in the
Digital Credential ecosystem.

## Validity Reasons

Status assertions can reveal information about the holder or subject that was not commited to in the original credential issuance.
Providing a reason that a digital credential is no longer valid can be essential to certian use cases, and unacceptable for others.

> For example, in a healthcare setting, a patient should not have have reasons for a suspended credential disclosed in assertions of suspension.

Author: ```suggestion
For example, in a healthcare setting, a patient should not have medical reasons for a suspended credential disclosed in assertions of suspension.
```
